From 18de79a20a9492884f2d6532a1da484bb24b1c55 Mon Sep 17 00:00:00 2001
Message-Id: <18de79a20a9492884f2d6532a1da484bb24b1c55.1427265330.git.chang-joon.lee@intel.com>
In-Reply-To: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
References: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 18 Mar 2015 22:30:16 +0530
Subject: [PATCH 16/22] SQUASHME! [VPG]: drm/i915: Fix power initialization in
 S3 resume

i915 can wake up either by HWC or other wake events in kernel. If wakeup
comes from other events we skip modeset based on DPMS State. Since we
dont do modeset we need make sure we release the power init done in
early resume to keep idle count in sync and allow i915 to go back to
sleep.

The fumble was introduced in:
commit 1d9aa058a32f5bf55608627f5beae30015987765
change-Id: Iaa30d56cc4dd78c78e2809f62ef1a0be37778f7d
Author: Deepak S <deepak.s@intel.com>
Date:   Wed Mar 11 16:51:30 2015 +0530

    FOR_UPSTREAM [VPG]: drm/i915: Simplified the logic for delaying
    Display power gating.

Issue: GMINL-7207
Change-Id: I12d3bd0a0654d5de3a13bc5ef1d77c3d19b9eec5
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 821cb9a..09db821 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -773,7 +773,8 @@ static int __i915_drm_thaw(struct drm_device *dev, bool restore_gtt_mappings)
 			drm_modeset_lock_all(dev);
 			intel_modeset_setup_hw_state(dev, true);
 			drm_modeset_unlock_all(dev);
-		}
+		} else
+			intel_display_set_init_power(dev_priv, false);
 
 		/*
 		 * ... but also need to make sure that hotplug processing
-- 
1.7.9.5

